### PR TITLE
Add P3A metrics for Android Privacy Hub report

### PR DESCRIPTION
**Pull request by DJAndries**
Resolves https://github.com/brave/brave-browser/issues/30623

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

For `Brave.PrivacyHub.IsEnabled`:

0. Start with fresh profile, ensure metric does not exist in local state.
1. Enable the privacy report
2. Ensure the metric exists with the value 1.
3. Disable the privacy report
4. Ensure the metric is removed from local state.

For `Brave.PrivacyHub.Views`:

0. Start with fresh profile, ensure metric does not exist in local state.
1. Enable and view the privacy report.
2. Ensure the metric exists with the value 0.
3. Test different metric values by opening the privacy report several times.
4. Advance system time by 31 days, do not access report. Ensure metric does not exist in local state.
5. Open privacy report, ensure metric reappears with correct value.

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/app/BraveActivity.java [hunk 1]
```diff
@@ -1996,6 +2003,16 @@ private void InitAssetRatioService() {
         mAssetRatioService = AssetRatioServiceFactory.getInstance().getAssetRatioService(this);
     }
 
+    private void InitPrivacyHubMetrics() {
```
Comment (deeppandya) [inline android/java/org/chromium/chrome/browser/app/BraveActivity.java:2006]: nit : method name should in camel case

File: android/java/org/chromium/chrome/browser/misc_metrics/PrivacyHubMetricsFactory.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    private MessagePipeHandle wrapNativeHandle(long nativeHandle) {
+        return CoreImpl.getInstance().acquireNativeHandle(nativeHandle).toMessagePipeHandle();
+    }
+
+    @NativeMethods
+    interface Natives {
+        long getInterfaceToPrivacyHubMetrics();
+    }
+}
```
Comment (deeppandya) [inline android/java/org/chromium/chrome/browser/misc_metrics/PrivacyHubMetricsFactory.java:51]: nit : new line at the end

File: android/java/org/chromium/chrome/browser/onboarding/OnboardingPrefManager.java [hunk 3]
```diff
@@ -181,6 +183,11 @@ public void setBraveStatsEnabled(boolean enabled) {
         SharedPreferences.Editor sharedPreferencesEditor = mSharedPreferences.edit();
         sharedPreferencesEditor.putBoolean(PREF_BRAVE_STATS, enabled);
         sharedPreferencesEditor.apply();
+        try {
+            BraveActivity activity = BraveActivity.getBraveActivity();
+            activity.getPrivacyHubMetrics().recordEnabledStatus(enabled);
+        } catch (BraveActivityNotFoundException e) {
+        }
```
Comment (deeppandya) [inline android/java/org/chromium/chrome/browser/onboarding/OnboardingPrefManager.java:190]: we should a log to capture the exception details.

File: android/java/org/chromium/chrome/browser/settings/BraveStatsPreferences.java [hunk 4]
```diff
@@ -13,17 +13,18 @@
 import androidx.preference.Preference.OnPreferenceChangeListener;
 
 import org.chromium.base.ContextUtils;
-import org.chromium.base.task.AsyncTask;
 import org.chromium.base.ThreadUtils;
+import org.chromium.base.task.AsyncTask;
 import org.chromium.chrome.R;
 import org.chromium.chrome.browser.BraveRelaunchUtils;
+import org.chromium.chrome.browser.app.BraveActivity;
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/settings/BraveStatsPreferences.java:20]: why has that imported been added?

File: android/java/org/chromium/chrome/browser/app/BraveActivity.java (hunk 1 above)
Comment (DJAndries) [reply android/java/org/chromium/chrome/browser/app/BraveActivity.java:2006]: right, i was copying the casing from the methods above this one

File: android/java/org/chromium/chrome/browser/misc_metrics/PrivacyHubMetricsFactory.java (hunk 2 above)
Comment (DJAndries) [reply android/java/org/chromium/chrome/browser/misc_metrics/PrivacyHubMetricsFactory.java:51]: i tried, but presubmit removes it

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Review by deeppandya — APPROVED** [review]
LGTM

File: components/misc_metrics/privacy_hub_metrics_unittest.cc [hunk 5]
```diff
@@ -0,0 +1,50 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
```
Comment (iefremov) [inline components/misc_metrics/privacy_hub_metrics_unittest.cc:1]: 2023

**Review by iefremov — APPROVED** [review]

**Pull request merged**